src/preprocess/links.rs

```diff
@@ -155,24 +182,63 @@ fn parse_include_path(path: &str) -> LinkType<'static> {
     let end = end.and_then(|s| s.parse::<usize>().ok());
     match start {
         Some(start) => match end {
-            Some(end) => LinkType::IncludeRange(path, Range { start, end }),
+            Some(end) => LinkType::IncludeRange(path, SomeRange::Range(Range { start, end })),
             None => {
                 if has_end {
-                    LinkType::IncludeRangeFrom(path, RangeFrom { start })
+                    LinkType::IncludeRange(path, SomeRange::RangeFrom(RangeFrom { start }))
                 } else {
                     LinkType::IncludeRange(
                         path,
-                        Range {
+                        SomeRange::Range(Range {
+                            start,
+                            end: start + 1,
+                        }),
+                    )
+                }
+            }
+        },
+        None => match end {
+            Some(end) => LinkType::IncludeRange(path, SomeRange::RangeTo(RangeTo { end })),
+            None => LinkType::IncludeRange(path, SomeRange::RangeFull(RangeFull)),
+        },
+    }
+}
+
+fn parse_rustpartial_path(path: &str) -> LinkType<'static> {
+    let mut parts = path.split(':');
+    let path = parts.next().unwrap().into();
+
+    let next_element = parts.next();
+    let start = if let Some(value) = next_element.and_then(|s| s.parse::<usize>().ok()) {
+        // subtract 1 since line numbers usually begin with 1
+        Some(value.saturating_sub(1))
+    } else {
+        None
+    };
+
+    let end = parts.next();
+    let has_end = end.is_some();
+    let end = end.and_then(|s| s.parse::<usize>().ok());
+    match start {
+        Some(start) => match end {
+            Some(end) => LinkType::RustPartial(path, SomeRange::Range(Range { start, end })),
+            None => {
+                if has_end {
```

Comment:
How can `has_end` be true if `end` is `None`? Seems like this is hiding user input error.

Reply:
If you want to keep it, can you maybe do `end.unwrap_or(...)`?